cfgov/jinja2/v1/_includes/molecules/related-posts.html

```diff
@@ -27,8 +26,20 @@ <h2 class="header-slug">
         </span>
     </h2>
     {% endif %}
-    {#% TODO: Add condition to display various related post types %#}
-    {{ _related_posts_list() }}
+    {% set related_post_types = page.related_posts %}
+    {% if related_post_types | length  > 1 %}
+    {% for post_type, post_dict in related_post_types.iteritems() %}
+        <div class="m-related-posts_list-containter">
+            <h3 class="h4">
+                {{ category_icon.render(post_type) or category_icon.render('blog') }}
+                {{ post_type | title }}
```

Comment:
The values of `post_type` will be `posts`, `newsroom`, and `events` so there should be a better way of rendering these names. I think I can take care of this in my PR.